### ISSUE TITLE
To Do Objectives

### ISSUE THREAD

**Issue by hegyak** [hegyak/SkyEditGUI#1]
To help keep track of what needs to be done:
- [x] Rewrite the Entire Load/Write Methods for Figures.  It's necessary because of the Three Types I am dealing with.
- [x] Determine if Usage Data (Time, Number of Owners) has any impact on Game.

- [x] Double check if the Giants Figure Skills has something unique going on.

- [x] See if Swap Bottom has Unique behavior Vs. Other Figures for Skills.

- Does a Swap Bottom Half even have any impact on game?  How?

- [ ] Figure out how a Trap's Special Checksum is done/setup.

- [ ] Finish Editor GUI and Functionality for Vehicles.

-   What are the Maximum Values for Weapons, Shields and Gearbits?

- What order are the Mods parsed/handled for Vehicles?

- [ ] Create Editor GUI for Crystals.

- There's a LOT of stuff to Set/Do for this one.

- Crystals have this special Checksums.  Similar to Trap Checksum

- [ ] When Changing Figure ID/Variant ID, Verify that it's not a Special Restricted Figure and Disable/Enable Controls as necessary.

Possible:
- [ ] Improve the GUI  (How?)
- [ ] Improve Figure Identification
- [ ] Verify the Gen6 Figures Hashes.

**Comment by hegyak** [comment]
The Usage data, has no real Impact on the game.

**Comment by hegyak** [comment]
The Games beyond Gen 1, care about Type 4 CRC or the Specialized replacement.

**Comment by hegyak** [comment]
Aside from Being a Bottom half, the bottom half of a Duo Figure, has no real game impact aside from Level.  Skills and other values are safe to ignore.

**Comment by hegyak** [comment]
Added:
Rewrite Load/Save Method to the list.

**Comment by hegyak** [comment]
Added Special Disable/Enable function handling for Figures that are Unique and don't behave as Normal Figures when the User selects from the ListBox.

**Comment by hegyak** [comment]
The Rewriting of Reading/Writing is a High Priority issue now.  ETA, Alpha 6 release.

**Comment by hegyak** [comment]
Figure IO has been rewritten.  It's now Unified as of Alpha 6.

**Comment by thatonecookie** [comment]
Do you plan on being enable to change weapons and body parts? and the level for a creation crystal character

**Comment by hegyak** [comment]
@thatonecookie Yes, that is on my Todo list.  Amongst many other things to do with this editor.  However, the Checksum for Crystals needs to be figured out, or the edited figure will not work in game.

**Comment by Peppins32** [comment]
Any news? Crystals?

**Comment by j-tools** [comment]
Hi, thank you for your work so far. I have quite an extensive Skylander collection so if needed, I can help you with a lot of raw data. You can email or DM me and tell me what I can do to support your efforts. I can also support with beta testing if you like

**Comment by JackShimada** [comment]
is this project still being worked on?
